### PR TITLE
SOLR-12226: Basic zk test framework in BATS

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-12226

# Description

Start testing ZK interaction as we get into cli changes..

# Solution

basic bats test to see if it works

# Tests
literally wrote a test ;-)

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by epugh** [comment]
I have a start here...   Including uncovering an interesting `bin/solr cp` command that looks aliased somehow, but not the same, to `bin/solr zk cp`

**Comment by epugh** [comment]
Okay, I've got upconfig and ls in here...   I think this is ready for review..  if there are some specific scearnious that you think need adding, please holler.

File: solr/packaging/test/test_zk.bats [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  rm myfile.txt
+  rm myfile2.txt
+}
+
+@test "upconfig" {
+  local source_configset_dir="${SOLR_TIP}/server/solr/configsets/sample_techproducts_configs"
+  test -d $source_configset_dir
+
+  run solr zk zk upconfig -d ${source_configset_dir} -n techproducts2 -z localhost:9983
```
Comment (risdenk) [inline solr/packaging/test/test_zk.bats:69]: `solr zk zk` - double zk?

**Review by risdenk — CHANGES_REQUESTED** [review]

File: solr/packaging/test/test_zk.bats (hunk 1 above)
Comment (epugh) [reply solr/packaging/test/test_zk.bats:69]: It's like doublemint gum...   it's twice the fun!

File: solr/packaging/test/test_zk.bats [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+@test "listing out files" {
+  sleep 1
+  run solr zk ls / -z localhost:9983
+  assert_output --partial "aliases.json"
+}
+
+@test "copying files around" {
+  touch myfile.txt
+  # Umm, what is solr cp?  It's like bin/solr zk cp but not?
+  run solr cp -src myfile.txt -dst zk:/myfile.txt -z localhost:9983
```
[the remark this answers is not in the document]
Comment (epugh) [reply solr/packaging/test/test_zk.bats:48]: I have added https://issues.apache.org/jira/browse/SOLR-16780 to track this.

**Pull request merged**